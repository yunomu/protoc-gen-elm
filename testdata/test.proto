syntax = "proto3";

package test;

option go_package = "./test";

message User {
  string user_id = 1;
  string user_name = 2;
  int32 age = 3;
}

message UserList {
  repeated User users = 1;
}
