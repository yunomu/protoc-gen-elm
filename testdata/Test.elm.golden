-- DO NOT EDIT
--
-- This file is automatically generated by protoc-gen-elm.
-- Source: test.proto

module Test.Test exposing (..)

import Json.Decode as Decode
import Json.Decode.Pipeline as Pipeline
import Json.Encode as Encode

type alias User =
    { 
    userId : String,
    userName : String,
    age : Int
    }

userDecoder : Decode.Decoder User
userDecoder =
    Pipeline.decode User
        |> Pipeline.required "userId" Decode.string
        |> Pipeline.required "userName" Decode.string
        |> Pipeline.required "age" Decode.int

userEncoder : User -> Encode.Value
userEncoder user =
    Encode.object
        [ 
            ("userId", Encode.string user.userId),
            ("userName", Encode.string user.userName),
            ("age", Encode.int user.age)
        ]

type alias UserList =
    { 
    users : (List User)
    }

userListDecoder : Decode.Decoder UserList
userListDecoder =
    Pipeline.decode UserList
        |> Pipeline.required "users" (Decode.list userDecoder)

userListEncoder : UserList -> Encode.Value
userListEncoder userList =
    Encode.object
        [ 
            ("users", (Encode.list userEncoder userList.users))
        ]

